### ISSUE TITLE
Installing Cyclone DDS Error ddsconf.h: No such file or directory

### ISSUE THREAD

**Issue by rahul19861986** [eclipse-cyclonedds/cyclonedds#841]
I was trying to install Cyclone  DDS on my windows 10  machine, to use with Eclipse. My build has gone successfull and got this message 
`-- Generating done
-- Build files have been written to: C:/Users/rkpsr/cyclonedds/build`

when i try to do the install
`$ cmake --build . --target install`

compilation is getting terminated of following error 

`C:\Users\rkpsr\cyclonedds\src\tools\ddsconf\ddsconf.c:30:10: fatal error: ddsconf.h: No such file or directory
   30 | #include "ddsconf.h"
      |          ^~~~~~~~~~~
compilation terminated.
`

how can i solve this issue? 

this is the complete result 

`rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake -G "Eclipse CDT4 - MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=/user -DBUILD_EXAMPLES=ON ..
-- Eclipse version is set to 3.6 (Helios). Adjust CMAKE_ECLIPSE_VERSION if this is wrong.
-- Building with OpenSSL support
-- Feature rusage disabled
-- Feature netstat disabled
-- Found BISON: C:/Program Files (x86)/GnuWin32/bin/bison.exe (found suitable version "3.7.4", minimum required is "3.0.4")
-- Configuring done
CMake Warning in CMakeLists.txt:
  The build directory is a subdirectory of the source directory.

  This is not supported well by Eclipse.  It is strongly recommended to use a
  build directory which is a sibling of the source directory.

-- Generating done
-- Build files have been written to: C:/Users/rkpsr/cyclonedds/build

rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake --build . --target install
[  0%] Building C object src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj
C:\Users\rkpsr\cyclonedds\src\tools\ddsconf\ddsconf.c:30:10: fatal error: ddsconf.h: No such file or directory
   30 | #include "ddsconf.h"
      |          ^~~~~~~~~~~
compilation terminated.
src\tools\ddsconf\CMakeFiles\ddsconf.dir\build.make:75: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj' failed
mingw32-make.exe[2]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj] Error 1
CMakeFiles\Makefile2:534: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/all' failed
mingw32-make.exe[1]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/all] Error 2
Makefile:154: recipe for target 'all' failed
mingw32-make.exe: *** [all] Error 2`

**Comment by k0ekk0ek** [comment]
Hi @rahul19861986! The step that was successful is actually just the configure step in which it generates the makefiles. The `cmake --build . --target install` command builds and installs (dropping `--target install` allows you to just build). MinGW builds are known to fail at this point in time. We've been meaning to get it supported but unfortunately haven't found the time to do so yet. I can give it a quick look, but if you want to get started on Windows quickly, using Visual Studio (Community) is your best bet.

**Comment by k0ekk0ek** [comment]
I just tried a quick MinGW build. I did not run into the issue where `ddsconf.h` is not found. Can you try with just the "MinGW Makefiles" generator?

**Comment by rahul19861986** [comment]
> I just tried a quick MinGW build. I did not run into the issue where `ddsconf.h` is not found. Can you try with just the "MinGW Makefiles" generator?

Hi
I'm new to this dds and MinGW can you point me to any tutorial?

**Comment by rahul19861986** [comment]
Hello 
I tried just using "MinGW Makefiles", still it says terminated and same error 

```
rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$  cmake -G "MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=/user -DBUILD_EXAMPLES=ON ..
-- Eclipse version is set to 3.6 (Helios). Adjust CMAKE_ECLIPSE_VERSION if this is wrong.
-- Building with OpenSSL support
-- Feature rusage disabled
-- Feature netstat disabled
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/rkpsr/cyclonedds/build

rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake --build
Usage: cmake --build [<dir> | --preset <preset>] [options] [-- [native-options]]
Options:
  <dir>          = Project binary directory to be built.
  --preset <preset>, --preset=<preset>
                 = Specify a build preset.
  --list-presets
                 = List available build presets.
  --parallel [<jobs>], -j [<jobs>]
                 = Build in parallel using the given number of jobs.
                   If <jobs> is omitted the native build tool's
                   default number is used.
                   The CMAKE_BUILD_PARALLEL_LEVEL environment variable
                   specifies a default parallel level when this option
                   is not given.
  --target <tgt>..., -t <tgt>...
                 = Build <tgt> instead of default targets.
  --config <cfg> = For multi-configuration tools, choose <cfg>.
  --clean-first  = Build target 'clean' first, then build.
                   (To clean only, use --target 'clean'.)
  --verbose, -v  = Enable verbose output - if supported - including
                   the build commands to be executed.
  --             = Pass remaining options to the native tool.

rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake --build .
[  0%] Building C object src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj
C:\Users\rkpsr\cyclonedds\src\tools\ddsconf\ddsconf.c:30:10: fatal error: ddsconf.h: No such file or directory
   30 | #include "ddsconf.h"
      |          ^~~~~~~~~~~
compilation terminated.
src\tools\ddsconf\CMakeFiles\ddsconf.dir\build.make:75: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj' failed
mingw32-make.exe[2]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj] Error 1
CMakeFiles\Makefile2:534: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/all' failed
mingw32-make.exe[1]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/all] Error 2
Makefile:154: recipe for target 'all' failed
mingw32-make.exe: *** [all] Error 2
```

**Comment by k0ekk0ek** [comment]
Judging by the first line of output (`-- Eclipse version is set to 3.6 (Helios). ...`), I think you didn't clear out the build directory before reconfiguring(?) Please do so first and the rerun the CMake commands.

**Comment by rahul19861986** [comment]
i have cleaned and tried again, but it still says eclipse version 

```
rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake --build  C:/Users/rkpsr/cyclonedds/build --target clean

rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake -G "MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=/user -DBUILD_EXAMPLES=ON ..
-- Eclipse version is set to 3.6 (Helios). Adjust CMAKE_ECLIPSE_VERSION if this is wrong.
-- Building with OpenSSL support
-- Feature rusage disabled
-- Feature netstat disabled
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/rkpsr/cyclonedds/build

rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$  cmake --build
Usage: cmake --build [<dir> | --preset <preset>] [options] [-- [native-options]]
Options:
  <dir>          = Project binary directory to be built.
  --preset <preset>, --preset=<preset>
                 = Specify a build preset.
  --list-presets
                 = List available build presets.
  --parallel [<jobs>], -j [<jobs>]
                 = Build in parallel using the given number of jobs.
                   If <jobs> is omitted the native build tool's
                   default number is used.
                   The CMAKE_BUILD_PARALLEL_LEVEL environment variable
                   specifies a default parallel level when this option
                   is not given.
  --target <tgt>..., -t <tgt>...
                 = Build <tgt> instead of default targets.
  --config <cfg> = For multi-configuration tools, choose <cfg>.
  --clean-first  = Build target 'clean' first, then build.
                   (To clean only, use --target 'clean'.)
  --verbose, -v  = Enable verbose output - if supported - including
                   the build commands to be executed.
  --             = Pass remaining options to the native tool.

rkpsr@Rahul MINGW64 ~/cyclonedds/build (master)
$ cmake --build .
[  0%] Building C object src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj
C:\Users\rkpsr\cyclonedds\src\tools\ddsconf\ddsconf.c:30:10: fatal error: ddsconf.h: No such file or directory
   30 | #include "ddsconf.h"
      |          ^~~~~~~~~~~
compilation terminated.
src\tools\ddsconf\CMakeFiles\ddsconf.dir\build.make:75: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj' failed
mingw32-make.exe[2]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj] Error 1
CMakeFiles\Makefile2:534: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/all' failed
mingw32-make.exe[1]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/all] Error 2
Makefile:154: recipe for target 'all' failed
mingw32-make.exe: *** [all] Error 2
```

**Comment by rahul19861986** [comment]
since it was not clearing properly, i have created a new folder called build 2 and tried with only "MinGW Makefiles"
Still i get same error , am i missing something?


```
rkpsr@Rahul MINGW64 ~
$ cd cyclonedds

rkpsr@Rahul MINGW64 ~/cyclonedds (master)
$ mkdir build2

rkpsr@Rahul MINGW64 ~/cyclonedds (master)
$ cd build2

rkpsr@Rahul MINGW64 ~/cyclonedds/build2 (master)
$ cmake -G "MinGW Makefiles" -DCMAKE_INSTALL_PREFIX=/user -DBUILD_EXAMPLES=ON ..
-- The C compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/cygwin64/bin/gcc.exe
-- Check for working C compiler: C:/cygwin64/bin/gcc.exe - works
-- Detecting C compile features
-- Detecting C compile features - done
-- Setting build type to 'RelWithDebInfo' as none was specified.
CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.20/Modules/GNUInstallDirs.cmake:236 (message):
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  CMakeLists.txt:258 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY) (found version "1.1.1k")
-- Building without OpenSSL support
-- Check target architecture - done
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of intmax_t
-- Check size of intmax_t - done
-- Looking for stpcpy
-- Looking for stpcpy - found
-- Looking for strlcpy
-- Looking for strlcpy - found
-- Looking for strlcat
-- Looking for strlcat - found
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Feature rusage disabled
-- Feature netstat disabled
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found BISON: C:/Program Files (x86)/GnuWin32/Bison/win_bison.exe (found suitable version "3.7.4", minimum required is "3.0.4")
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/rkpsr/cyclonedds/build2

rkpsr@Rahul MINGW64 ~/cyclonedds/build2 (master)
$ cmake --build .
[  1%] Building C object src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj
C:\Users\rkpsr\cyclonedds\src\tools\ddsconf\ddsconf.c:30:10: fatal error: ddsconf.h: No such file or directory
   30 | #include "ddsconf.h"
      |          ^~~~~~~~~~~
compilation terminated.
src\tools\ddsconf\CMakeFiles\ddsconf.dir\build.make:75: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj' failed
mingw32-make.exe[2]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/ddsconf.c.obj] Error 1
CMakeFiles\Makefile2:447: recipe for target 'src/tools/ddsconf/CMakeFiles/ddsconf.dir/all' failed
mingw32-make.exe[1]: *** [src/tools/ddsconf/CMakeFiles/ddsconf.dir/all] Error 2
Makefile:154: recipe for target 'all' failed
mingw32-make.exe: *** [all] Error 2
```

**Comment by k0ekk0ek** [comment]
Hi @rahul19861986, don't know if you're still interested in support for MinGW. But #842 adds support for building with it. It's not on the master branch yet and it very likely won't be merged before the 0.8.0 release, but you can give it a try if you want.

**Comment by rahul19861986** [comment]
I'm really grateful for the support you are offering
I will surely try it

**Comment by k0ekk0ek** [comment]
Closed via #842.
